PluginCode/src/main/java/smr/cs/ualberta/libcomp/action/ReplacementAction.java

```diff
@@ -209,51 +222,65 @@ public void detectDependancy(@NotNull final Editor editor, @NotNull final PsiFil
-                    if (choicesArray.size() > 0) {
-                        DependencyStatement depObj = new DependencyStatement();
-                        depObj.setImportLocation(i);
-                        depObj.setImportLib(Integer.parseInt(choicesArray.get(0)));
-                        depObj.setImportDomain(Integer.parseInt(choicesArray.get(1)));
-                        depObj.setDomainName(choicesArray.get(2));
-                        depObj.setFromlocation(startOffset);
-                        depObj.setTolocation(endOffset);
-
-                        if (dataAccessObject.isEnabled(Integer.parseInt(choicesArray.get(1)), project_name)) {
-                            depObj.setEnableddomain(false);
-                        }
-                        else {
-                            depObj.setEnableddomain(true);
-                        }
+                selectedTerm = valuesInQuotes[0];
 
-                        DependListObjects.add(depObj);
+                DatabaseAccess dataAccessObject = new DatabaseAccess();
+                ArrayList<String> choicesArray = dataAccessObject.selectJsonAllLibraries(selectedTerm);
 
+                if (choicesArray.size() > 0) {
+                    DependencyStatement depObj = new DependencyStatement();
+                    depObj.setImportLocation(i);
+                    depObj.setImportLib(Integer.parseInt(choicesArray.get(0)));
+                    depObj.setImportDomain(Integer.parseInt(choicesArray.get(1)));
+                    depObj.setDomainName(choicesArray.get(2));
+                    depObj.setFromlocation(startOffset);
+                    depObj.setTolocation(endOffset);
 
-                        if (depObj.getEnableddomain()) {
-                            editorModel.addLineHighlighter(i,
-                                    DebuggerColors.BREAKPOINT_HIGHLIGHTER_LAYER + 1, softerAttributes);
+                    if (dataAccessObject.isEnabled(Integer.parseInt(choicesArray.get(1)), projectName)) {
+                        depObj.setEnableddomain(false);
+                    }
+                    else {
+                        depObj.setEnableddomain(true);
+                    }
 
-                        }
+                    listObjects.add(depObj);
 
+                    if (depObj.getEnableddomain()) {
+                        editorModel.addLineHighlighter(i,
+                                DebuggerColors.BREAKPOINT_HIGHLIGHTER_LAYER + 1, softerAttributes);
 
                     }
                 }
+            }
             ++i;
-            boolean isContainsEnd = lineText.contains("}");
+            boolean isContainsEnd = lineText.contains(endIdentifier);
             if (isContainsEnd) {
                 dependenciesExists = false;
             }
         }
     }
 
+    public void detectDependancy(@NotNull final Editor editor, @NotNull final PsiFile psiFile, @NotNull final  String projectName ) throws IOException {
```

Comment:
can you check spelling of dependency throughout the code?

Reply:
sure. thank you